backend/climateconnect_api/utility/notification.py

```diff
@@ -33,9 +43,43 @@ async def send_out_live_notification(user_id):
     await channel_layer.group_send("user-" + str(user_id), {"type": "notification"})
 
 
+def create_follower_notification(
+    notif_type_number, # Integer value of the notification type
+    look_up_follower_type_field_name, # name of the field being looked up based on the follower type
+    look_up_entitiy_type_field_name, # name of the field being looked up based on the model type
+    member_type_model, # type of team to be looked at when getting their members (e.g project or org team)
+    follower_type, # type of follower e.g project or org
+    follower_type_entity, # type of entity being followed e.g. org or project
```

Comment:
I'm not too sure if I understand this correctly. Within the functions that call the functon `create_follower_notification`,  we are deriving the necessary info based on which follower type is given. We could also derive them within  `create_follower_notification` but that would defeat the purpose of a generic function?

Reply:
functions calling `create_follower_notification` being `create_project_follower_notification` and  `create_organization_follower_notification`. And it is here where the necessary information is derived.

Reply:
Just looked at this again. Actually some of the values are pretty misleading :D 
1. If we know `notif_type_number` we can derive the values of `look_up_follower_type_field_name`, `look_up_entitiy_type_field_name` and `member_type_model` with that. Therefore we should not pass those down as variables because it adds unnecessary complexity. Instead we could use a helper function along the lines of this screenshot
![Screenshot from 2022-11-15 07-56-25](https://user-images.githubusercontent.com/10531079/201850752-899e738a-f6ce-4c0c-9be3-4024ff572934.png)

2. The variables `follower_type`, `follower_type_entity` and `follower_type_user_id` are pretty misleading.
`follower_type` is actually the follower object/instance. It could simply be called `follower` It's the ProjectFollower or OrganizationFollower object, not just the type. Also the comment is currently quite misleading here.
`follower_type_entity` makes me expect the value to be "project" or "organization" but in fact it's the specific organization the user followed. `follower_entity` as the name and an updated comment  would be better here
`follower_type_user_id` is simply the follower_user_id. I think in the comment here we can lose the ` who is a certain type (project/org)`.